KODI: Update NEON support

C-Prime90 opened the pull request:
## Description
This commit makes CCPUInfo::HasNeon() report NEON enabled on Aarch64 targets and updates the ifdef's for newer preprocessor.

## How Has This Been Tested?
This has bin tested locally on Amlogic S812/S905 running on EmbER firmware.

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

stefansaraev commented:
are you 100% that enabling -funsafe-math-optimizations is safe for kodi ?

croniccorey commented:
I've had it enabled in my Linux builds since 16.0 without issues, i can remove that if you'd prefer.

stefansaraev commented on cmake/scripts/linux/ArchSetup.cmake:
```diff
@@ -16,10 +16,11 @@ else()
     set(ARCH aarch64)
-    set(NEON False)
+    set(NEON True)
+    set(NEON_FLAGS "-march=armv8-a -funsafe-math-optimizations")
```
aarch64 is always neon. you dont need these, and you should not override user's specified -march= anyway

croniccorey commented:
I've updated the PR based on your comments.

MartijnKaijser commented:
@FernetMenta @peak3d ping as well

FernetMenta commented:
What non-NEON systems do we support? I am wondering if we can just drop the ifdef and make it default.

stefansaraev commented:
@FernetMenta rpi1/0

FernetMenta commented:
so we enable neon with a compiler switch, and then test for preprocessor macros? seems a bit combersome and error prone to me. finally results in this kind of issues.
why don't we user something like HAS_NEON that is set by defult and can be disabled for the outdated hardware

wsnipex commented:
agreed, we should be using the preprocessor defines here.

MartijnKaijser commented:
@croniccorey could you do the requested changes?

croniccorey commented:
so you would like to drop the compiler switch in favor of preprocessor defines?

wsnipex commented:
>You would like to drop the compiler switch in favor of preprocessor defines or drop preprocessor defines in favor of HAS_NEON set by cmake?
yes, this please

croniccorey commented:
Ok i will update PR to use HAS_NEON

C-Prime90 committed:
[Kodi] Update NEON support

- Define HAS_NEON when NEON cmake switch is set
- Aarch64 always has neon, Enable cmake switch and change CPUInfo to report it
- Don't set NEON_FLAGS if none exist, Not all targets need extra flags
- Don't use preprocessor to check for neon support, Instead use HAS_NEON set by cmake

croniccorey commented:
@MartijnKaijser @FernetMenta @wsnipex I have updated the PR based on your comments

wsnipex commented:
jenkins build this please

MartijnKaijser commented:
@croniccorey thx for sticking with us and for your contribution
